### PR TITLE
ci: bump neovim and tree-sitter cli to latest version

### PR DESCRIPTION
**Pull request by clason**
possibly want to wait to update to tree-sitter 0.21.0 (or whatever the upcoming version is)

also simplify tests:
* only run on push to master _or_ PR opened/synchronized
* cancel in-progress workflows for updated PRs
* only run ubuntu-latest+gcc+nightly on draft PRs
* only run ubuntu+gcc and macos+clang (plus windows x 3)

### CODE REVIEW

**Comment by theHamsta** [comment]
Probably you can reduce to one compiler on Ubuntu/Mac

**Comment by clason** [comment]
Historically, what were the outliers (platform+compiler combinations that failed even when the others passed)?

File: .github/workflows/check-query-files-and-compilation.yml
Comment (theHamsta) [inline .github/workflows/check-query-files-and-compilation.yml:55]: I'm a bit worried about this file getting too complicated by this copy and paste. I'd rather just kick out configuration and cache parser installation

Comment (clason) [reply .github/workflows/check-query-files-and-compilation.yml:55]: I think it's still fine; Neovim does the same. (If there's a better way of skipping workflows inside a matrix job, I'm all ears, though.)

Comment (clason) [reply .github/workflows/check-query-files-and-compilation.yml:55]: (I thought about simplifying the draft job by hardcoding everything, but that seemed even more error-prone.)

**Comment by clason** [comment]
The biggest bottleneck is the three Windows compilers. If we can reduce that, that would be nice.

**Comment by theHamsta** [comment]
@clason parallel compilation speeds up compilation on MacOS/Linux by huge factor but on Windows it will cause a crash due to OOM on the VM :smile:  when using MSVC. All in all, the CI is not a huge issue 8 minutes aren't too bad with auto-merge.

**Comment by clason** [comment]
Windows clang takes 22 minutes...

**Comment by clason** [comment]
Anyway, I'd like to merge this ASAP so we can build on top of it. I can remove the special case draft-only job if you wish, if that is what is blocking this?

File: .github/workflows/check-query-files-and-compilation.yml
Comment (MunifTanjim) [inline .github/workflows/check-query-files-and-compilation.yml:1]: Why are the artifacts uploaded in this job? Are they used for something?

Comment (clason) [reply .github/workflows/check-query-files-and-compilation.yml:1]: I don't know; I believe there might be people (or workflows) who don't want to build their own parsers?

Comment (MunifTanjim) [reply .github/workflows/check-query-files-and-compilation.yml:1]: Doing these on each commit seems like a waste 🤔 

image

Comment (clason) [reply .github/workflows/check-query-files-and-compilation.yml:1]: Agreed.

Comment (MunifTanjim) [reply .github/workflows/check-query-files-and-compilation.yml:1]: Anyway, it's not related to this PR. I was just curious.

Comment (theHamsta) [reply .github/workflows/check-query-files-and-compilation.yml:1]: It was mostly for debugging because sometimes the CI would build successfully artifacts that wouldn't work. The Windows wiki has a hint that users can download the parsers from CI. But it would probably be better to have binary releases for that.

Comment (clason) [reply .github/workflows/check-query-files-and-compilation.yml:1]: Let me know if I should remove that step.

Comment (theHamsta) [reply .github/workflows/check-query-files-and-compilation.yml:1]: >Doing these on each commit seems like a waste thinking

Well, we're not paying the servers :smile: was wondering how much data we caused by that. They seem to be persistent for a pretty long time. I was expecting that at some point we'd either hit API limits or that stuff get's quickly ejected.

Comment (theHamsta) [reply .github/workflows/check-query-files-and-compilation.yml:1]: Probably, they should be replaced by releases. I was experimenting a bit with that https://github.com/theHamsta/nvim-treesitter/releases/tag/nightly-parsers releases would also have a predictable path for download (e.g. tagged releases of our plugin could auto-download the binaries for the platform if a user opts in). But even better would be to do that outside of our org.

Comment (MunifTanjim) [reply .github/workflows/check-query-files-and-compilation.yml:1]: > Well, we're not paying the servers

I meant "waste" as in "doing unnecessary work repeatedly as it is producing and uploading the same files for each commit" 😂 I'm sure github removes old files once it hits the limit.

Comment (theHamsta) [reply .github/workflows/check-query-files-and-compilation.yml:1]: or ghcr.io ?

Comment (clason) [reply .github/workflows/check-query-files-and-compilation.yml:1]: (+1 on releases, but not sure why outside of the org would be better?)

Comment (theHamsta) [reply .github/workflows/check-query-files-and-compilation.yml:1]: Because tree-sitter 1.0 wants to offer an easy/standardized way for parsers to publish without having generated files in their repo. But sure for us it would make sense to have all PRs

Comment (clason) [reply .github/workflows/check-query-files-and-compilation.yml:1]: Oh, I see. Yes, if upstream took care of this, that'd be cake :)

Comment (theHamsta) [reply .github/workflows/check-query-files-and-compilation.yml:1]: @clason See here how releases would work https://github.com/theHamsta/nvim-treesitter/commit/a0e29464e75bc1cf146523379d2d274c6f09d4b6 . You can only do this on tags because there you can hit API limits fairly quickly :smile:

Comment (clason) [reply .github/workflows/check-query-files-and-compilation.yml:1]: Yeah, we would probably want to do a daily job or something for that.

Anyway, all that is for a different PR -- is there anything I need to change for this one? I'm just waiting for your approval ;)

File: .github/workflows/check-query-files-and-compilation.yml
Comment (theHamsta) [inline .github/workflows/check-query-files-and-compilation.yml:24]: This shouldn't be nightly. Our main CI is latest stable on Ubuntu. Nightly is not always available for download and is therefore allowed to fail

Comment (theHamsta) [reply .github/workflows/check-query-files-and-compilation.yml]: It is also very annoying to change the required builds. You won't be able to merge this and I need to change it manually after force-merging this PR

Comment (clason) [reply .github/workflows/check-query-files-and-compilation.yml]: OK, so you _do_ want me to drop the fast draft CI -- you just have to say so!

**Review by theHamsta — APPROVED** [review]
Nice improvement! Feeling bad for having done this before...

**Comment by clason** [comment]
No worries! @MunifTanjim now you can go ahead :)

**Comment by theHamsta** [comment]
@clason this PR changed the required test names. All PRs will have to rebase now to be able to get merged!

**Comment by clason** [comment]
I don't think it did? Or was the `0.6.0` part of the name?

(In this case, we should switch to using the `stable` tag...)

**Pull request merged**
